test/common/Client.js

```diff
@@ -130,6 +130,14 @@ describe('The Client class', function() {
 
   });
 
+  describe('The withCustomMailboxRoles method', function() {
+    it('should allow custom role mailbox after initializing client with custom role', function() {
```

Comment:
what's the effect of defining these roles? I could make this test pass with a method doing `return this`.
You miss some tests about the behavior it adds to the mailboxes.

Reply:
We can, for example, assert that this is equal to the mailboxRole object we expect?

Reply:
I changed to expecting Mailbox role to be a certain value.